implementation_guides/javascript.md

```diff
@@ -2,13 +2,119 @@
+
+      scripts: {
+        ...
+        "pactTest":
+        "export NODE_ENV=pactTest && jest --testRegex \"/*(.test.pact.js)\" --runInBand --setupFiles ./pactSetup.js --setupTestFrameworkScriptFile ./pactTestWrapper.js"
+      }
+
+You'll also need to give Pact some metadata about your setup. This is done with `pactSetup.js`:    
+
+    const path = require('path');
+    const { Pact } = require('@pact-foundation/pact');
+
+    global.port = 8989;
+    global.provider = new Pact({
+      port: global.port,
+      log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+      dir: path.resolve(process.cwd(), 'pacts'),
+      spec: 2,
+      cors: true,
+      pactfileWriteMode: 'update',
+      consumer: /* the name of your consumer */,
+      provider: /* the name of your provider */
+    });
+
+ Global setup for Jest is handled with `pactTestWrapper.js`:
+
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000; //* This is to give the pact mock server time to start
+
+    beforeAll(done => {
+      provider.setup() // this sets up the server ready for interactions
+      .then(() => done());
+    });
+
+    afterAll(done => {
+      provider
+        .finalize()  // This tears down the server
+        .then(provider.verify) // This makes sure there are no interactions sneaking in after the last test
+        .then(done);
+    });
+
+Then you can write tests of the form `<filename>.test.pact.js`:
+
+    import { Pact } from '@pact-foundation/pact';
+    import { api } from /* wherever */;   // This is your client-side API layer
+
+
+    describe('The API', () => {
+      let url = 'http://localhost';
+
+      const EXPECTED_BODY = {
+
+      // Copy this block once per interaction under test
+      describe(/* The API interaction being tested in words (string) */, () => {
+        beforeEach(() => {
+          const interaction = {
+            uponReceiving: /* Describe the request in words (string) */,
+            withRequest: {
+              method:  /* 'GET' or 'POST' or whatever (string) */,
+              path:  /* '/foo/bar' (string) */,
+              query:  /* '?query=parameters' (string) */,
+              headers: {
+                Accept: 'application/json'
+                /* etc */
+              }
+            },
+            willRespondWith: {
+              status: 200,
+              headers: {
+                'Content-Type': 'application/json'
+                /* etc */
+              },
+              body: /* describe the body (object using the Pact DSL)*/
+            }
+          };
+          return provider.addInteraction(interaction);
+        });
+
+        // add expectations
+        it(/* describe the test */, done => {
+          api(url).someCall()
+            .then(response => {
+              expect(response).toEqual(/* check the response here, using the default values provided to the Pact DSL */);
+            })
+            .then(() => provider.verify())
```

Comment:
It's probably OK here, but as discussed in Slack, I like separating this from the test itself

Reply:
I'm pretty sure `afterEach(() => provider.verify())`  should work.

Reply:
Good catch.